### PR TITLE
Fixes runtime errors that occur when mindless mobs examine heirlooms, + code improvements

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

Mindless mobs (such as observers without an initialized mind) examining family heirlooms would cause runtime errors due to the `user.mind.has_antag_datum(...)` line. 

- Rearranges the examine proc to prevent runtimes from occurring due to mindless mobs examining heirlooms. Adds an observer check at the beginning, just for sanity sake, even though it doesn't matter functionally. 
- Renamed the proc to `on_examine` for clarity
- Auto doc'd the proc + comments
- Handled a reference on Destroy

## Why It's Good For The Game

Runtimes that stop examine bad

## Changelog

:cl: Melbert
fix: Fixes uncommon runtimes that could occur when observers examine heirlooms
/:cl:

### CODE REVIEW

File: code/datums/components/heirloom.dm [hunk 1]
```diff
@@ -9,16 +12,30 @@
 	owner = new_owner
 	family_name = new_family_name
 
-	RegisterSignal(parent, COMSIG_PARENT_EXAMINE, .proc/examine)
+	RegisterSignal(parent, COMSIG_PARENT_EXAMINE, .proc/on_examine)
 
-/datum/component/heirloom/proc/examine(datum/source, mob/user, list/examine_list)
+/datum/component/heirloom/Destroy(force, silent)
+	owner = null
```
Comment (LemonInTheDark) [inline code/datums/components/heirloom.dm:18]: isn't strictly needed, but s a good pattern

File: code/datums/components/heirloom.dm [hunk 2]
```diff
@@ -9,16 +12,30 @@
...
+		examine_list += span_notice("It is the [family_name] family heirloom, belonging to [owner].")
+		return
 
 	if(user.mind == owner)
 		examine_list += span_notice("It is your precious [family_name] family heirloom. Keep it safe!")
-	else if(creeper && creeper.trauma.obsession == owner)
+		return
+
+	var/datum/antagonist/obsessed/our_creeper = user.mind.has_antag_datum(/datum/antagonist/obsessed)
+	if(our_creeper?.trauma.obsession == owner)
```
Comment (LemonInTheDark) [inline code/datums/components/heirloom.dm:38]: My gut says that it'd be cleaner to store the user's mind in a var, use ?. for it, and move the observer check to the creeper line, rather then duping the code

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]

File: code/datums/components/heirloom.dm (hunk 1 above)
Comment (MrMelbert) [reply code/datums/components/heirloom.dm:18]: better to have and not need than need than not have, I suppose

File: code/datums/components/heirloom.dm (hunk 2 above)
Comment (MrMelbert) [reply code/datums/components/heirloom.dm]: Sure thing, lemmie know if that's what you had in mind

Comment (LemonInTheDark) [reply code/datums/components/heirloom.dm]: 👍

**Pull request merged**